src/hooks/useGlobalAppState.ts

```diff
@@ -142,6 +77,29 @@ const onStateChange = (
  */
 export const useGlobalAppState = (): void => {
   const client = useClient()
+  const initFlag = useRef(false)
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  const routeIndex = useNavigationState(state => state?.index)
+
+  // On app start
+  useEffect(() => {
+    log.info('useGlobalAppState: app start')
+
+    if (getIsSecurityFlowPassed() && !initFlag.current) {
+      log.info('useGlobalAppState: app start, security flow passed')
+
+      if (isSharingMode()) {
+        log.info('useGlobalAppState: app start, sharing mode')
+        navigate(routes.sharing)
+      } else {
+        log.info('useGlobalAppState: app start, not sharing mode')
+      }
+
+      initFlag.current = true
```

Comment:
Seems like you never revert `initFlag.current` to `false`. Does this mean that this effect can happens only one single time? What happens if the apps goes to background, locks, then receive a sharing, multiple times in the same app lifecycle?

Reply:
the sharing flow is discarded as soon as the app goes into background/closes. At least I understood the spec like that